### ISSUE TITLE
pycdf argument types for M1 Mac

### ISSUE THREAD

**Issue by jtniehof** [spacepy/spacepy#546]
The M1 Mac uses a different convention for variadic functions and the resulting interaction with ctypes is...weird. The result is that any call to the CDF library `CDFlib` function fails.

This is described in [Python#42880](https://bugs.python.org/issue42880) (which also has some stuff that might be useful for general ctypes troubleshooting on M1). The upshot appears to be that the first argument *must* have a type specified. There are a few other pure-variadic functions, notably EPOCHbreakdown. This may be more complicated, as the first argument is a double. I may have to type-pun this function like in #20 (which reminds me to also look at CDF_TT2000_from_UTC_parts).

### Minimal example to reproduce issue:
```python
import spacepy.pycdf
```

### Error message/Traceback:
See #545, but that's complicated by the actual error not being thrown. What should really get thrown is `CDFError: BAD_FNC_OR_ITEM: The specified function or item is illegal.`

### OS, Python version, and dependency version information:
Python 3.9, M1 Mac

### Version of SpacePy
Current `master`

**Comment by selasley** [comment]
same issue with python 3.10.4 from python.org and the universal cdflib 3.8.1 on an M1 Mac 
```
$ file $CDF_LIB/libcdf.3.8.1.dylib 
/Applications/cdf/cdf38_1-dist/lib/libcdf.3.8.1.dylib: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64
- Mach-O 64-bit dynamically linked shared library x86_64] [arm64:Mach-O 64-bit dynamically linked shared library arm64
- Mach-O 64-bit dynamically linked shared library arm64]
/Applications/cdf/cdf38_1-dist/lib/libcdf.3.8.1.dylib (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
/Applications/cdf/cdf38_1-dist/lib/libcdf.3.8.1.dylib (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64

$ python3
Python 3.10.4 (v3.10.4:9d38120e33, Mar 23 2022, 17:29:05) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import spacepy.pycdf as CDF
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/me/venvs/py310/lib/python3.10/site-packages/spacepy/pycdf/__init__.py", line 1290, in <module>
    lib = Library(_libpath, _library)
  File "/Users/me/venvs/py310/lib/python3.10/site-packages/spacepy/pycdf/__init__.py", line 301, in __init__
    self.call(const.GET_, const.LIB_VERSION_, ctypes.byref(ver),
  File "/Users/me/venvs/py310/lib/python3.10/site-packages/spacepy/pycdf/__init__.py", line 641, in call
    return self.check_status(self._library.CDFlib(
  File "/Users/me/venvs/py310/lib/python3.10/site-packages/spacepy/pycdf/__init__.py", line 594, in check_status
    raise CDFError(status)
NameError: name 'CDFError' is not defined. Did you mean: 'EOFError'?

```

**Comment by jtniehof** [comment]
Test machine sys.platform is `'darwin'` and os.uname is `posix.uname_result(sysname='Darwin', nodename='phys-133258', release='20.6.0', version='Darwin Kernel Version 20.6.0: Tue Feb 22 21:10:42 PST 2022; root:xnu-7195.141.26~1/RELEASE_ARM64_T8101', machine='arm64')`

State of C functions with current ad-hoc test code:

| Function | Status |
| -------- | ------ |
| breakdownTT2000 | Crash | 
| CDF_TT2000_from_UTC_EPOCH | Good | 
| CDF_TT2000_from_UTC_EPOCH16 | Good | 
| CDF_TT2000_to_UTC_EPOCH | Good | 
| CDF_TT2000_to_UTC_EPOCH16 | Good | 
| CDFlib | Wrong results | 
| CDFsetFileBackward | Good | 
| computeEPOCH | Good | 
| computeEPOCH16 | Good | 
| computeTT2000 | Wrong results, but not yet doing ARM type-punning | 
| EPOCH16breakdown | Good | 
| EPOCHbreakdown | Good |

**Comment by jtniehof** [comment]
Ad-hoc test now works on 64-bit M1; doesn't work on 32-bit Pi; 64-bit Pi pending.

**Comment by xbonnin** [comment]
Just for information I had the same issue than selasley on my M1 Mac, but with Python 3.10.4, when installing via `pip install spacepy==0.3.0`
Installing from the spacepy github repo via `pip install -r requirements.txt && python setup.py install` command works well for me.

N.B. For some unknown reasons, spacepy works well when installation is done via pip under Python 3.8.10

**Comment by jtniehof** [comment]
> Just for information I had the same issue than selasley on my M1 Mac, but with Python 3.10.4, when installing via `pip install spacepy==0.3.0`

This fix is in 0.4.0 (in preparation).
